app/src/play/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -1640,30 +1640,25 @@ public void onErrorResponse(VolleyError error) {
     @Override
     public void onConnectionSuspended(int i) {
         Log.d("G+", "Connection suspended");
-        new Thread(() -> {
-            if (mGoogleApiClient != null)
-                mGoogleApiClient.connect();
-        }).run();
+        if (mGoogleApiClient != null)
+            mGoogleApiClient.connect();
```

Comment:
If this does any networking it will crash, have you tested?

Reply:
I am unable to do it , couple of files are missing like "BaseFile" etc many Files has been renamed (with still old import), although as i mentioned during previous commit 

invoke run won't create a new thread it just call the run function of runnable passed as lambda on UI thread so there is no threading with `.run();` if you want threading then it should use `.start();`

        new Thread(() -> {
            if (mGoogleApiClient != null)
                mGoogleApiClient.connect();
        }).start();
So currently lambda thread modules are running everything on UI.
Here is a [Small test that you can use to verify the behavior](https://gist.github.com/Pavneet-Sing/23bd944e322ebf02f32ac2cfa9c53fee)

Reply:
Change your build flavour to playDebug.
Maybe ´mGoogleApiClient.connect();´was never being reached so the exception never poped up, but if it would have run, it would have chashed; nevermind though, as connect() doesn't seem to make the connection from the thread it was called from.
Just a small question though, are you using AS 3.0?

Reply:
yes i am using 3.0 and if that's the case then you should close the PR , thanks for reply

Reply:
No need to close anything, it's OK.

Reply:
Ok good , Frankly, i was little free and I've been doing dry run (for fun but with caution ), i imported the project after your first comment and then created a little test ,  so if there is anything problematic then feel free to do anything , i''ll make the project up and running , keeping your suggestion  in mind :)